### PR TITLE
feat: Replace user graphql types with the generated types

### PR DESCRIPTION
**Pull request by peterstarling**
https://trello.com/c/D9LI7Syp/1610-3-replace-graphql-types-with-the-generated-ones-user

Updates some of the fixtures for other entities too

### CODE REVIEW

File: apps/asap-server/src/controllers/discover.ts [hunk 1]
```diff
@@ -98,7 +99,12 @@ export default class Discover implements DiscoverController {
     return {
       aboutUs: content.flatData.aboutUs || '',
       training: content.flatData.training?.map(parseGraphQLNewsAndEvents) ?? [],
-      members: content.flatData.members?.map(parseGraphQLUser) ?? [],
+      members:
+        content.flatData.members?.map((member) =>
+          parseGraphQLUser(
+            member as NonNullable<FetchUserQuery['findUsersContent']>,
```
Comment (kmaid) [inline apps/asap-server/src/controllers/discover.ts:105]: I was hoping to avoid casts like this. Is there no way around this?

**Review by kmaid — APPROVED** [review]

File: apps/asap-server/src/controllers/discover.ts (hunk 1 above)
Comment (peterstarling) [reply apps/asap-server/src/controllers/discover.ts:105]: i could add a @todo to finish this ticket https://trello.com/c/5vW6YMsL/1694-replace-graphql-types-with-the-generated-ones-news

File: apps/asap-server/src/controllers/users.ts [hunk 2]
```diff
@@ -293,26 +210,38 @@ export default class Users implements UserController {
...
+      return {
+        total: 0,
+        items: [],
+      };
+    }
+
     const { total, items } = queryUsersContentsWithTotal;
 
+    if (items === null) {
+      return {
```
[the remark this answers is not in the document]
Comment (peterstarling) [reply apps/asap-server/src/controllers/users.ts:228]: should we?

File: apps/asap-server/test/fixtures/research-output.fixtures.ts [hunk 3]
```diff
@@ -27,7 +30,7 @@ export const getSquidexResearchOutputGraphqlResponseAuthors = (): NonNullable<
     FetchResearchOutputQuery['findResearchOutputsContent']
   >['flatData']['authors']
 > =>
-  graphQlResponseFetchUsers.data.queryUsersContentsWithTotal.items.map(
+  getGraphqlResponseFetchUsers().data.queryUsersContentsWithTotal!.items!.map(
```
[the remark this answers is not in the document]
Comment (peterstarling) [reply apps/asap-server/test/fixtures/research-output.fixtures.ts:33]: well we can't cause some of them will be null

File: apps/asap-server/src/entities/user.ts [hunk 4]
```diff
@@ -61,41 +83,49 @@ export const parseGraphQLUserTeamConnection = (
...
-  const flatQuestions = item.flatData?.questions || [];
-  const flatSkills = item.flatData?.skills || [];
+    item.flatData.avatar || [];
+  const flatQuestions = item.flatData.questions || [];
+  const flatSkills = item.flatData.skills || [];
   const createdDate = parseDate(item.created).toISOString();
 
-  const role = ['Guest', 'Staff', 'Grantee'].includes(item.flatData?.role || '')
-    ? (item.flatData?.role as 'Guest' | 'Staff' | 'Grantee')
+  const role = ['Guest', 'Staff', 'Grantee'].includes(item.flatData.role || '')
```
[the remark this answers is not in the document]
Comment (peterstarling) [reply apps/asap-server/src/entities/user.ts:96]: yes

**Pull request merged**